### ISSUE TITLE
Command source not always updated when command can execute

### ISSUE THREAD

**Issue by TheJayMann** [Prolucid/Elmish.WPF#17]
After updating to the latest release and modifying my code to make use of command parameters rather than one way bindings updating the model, the command sources no longer check whether the command can execute when the UI changes (specifically, when the UI property bound to the CommandParameter changes).  Also, CommandManager.InvalidateRequerySuggested had no effect.  When I looked at the implementation of Command, I see that it has a CanExecuteChanged event implemented such that it is only raised when the view model is updated, as part of its notify function.  If it could be augmented to raise the event when the CommandManager.RequerySuggested event is raised, then it should allow UI changes which do not modify the view model to also update commmand sources.

**Comment by 2sComplement** [comment]
Thanks for raising this and for the corresponding PR. I'm curious to know how you're using `CommandParameter` since it was not meant to replace `Binding.cmd` or `Binding.oneWay`.  Are you calling `CommandManager.RequerySuggested` yourself or is it triggered by something else?

**Comment by TheJayMann** [comment]
So, first, I have a listbox defined as such:
```xaml
<ListBox ItemsSource="{Binding DataList}" x:Name="List"/>
```
Then I have a button defined as such:
```xaml
<Button Command="{Binding DataItemCommand}" CommandParameter={Binding SelectedItem, ElementName=List}">Do Something</Button>
```
Previously I would have the SelectedItem property of the ListBox one way to source bound to a property of the model, and the command would pull the property out of the model when executed, and also used this to determine if it could be executed.  With this style, I don't have to keep track of the currently selected item in the model, as the command source can determine this and push it into the CommandParameter, which is now available to the cmd functions.

The problem is that the custom commands being created were only raising the CanEcecuteChanged when the model itself was being changed, which meant that the button wasn't being made enabled when I selected an item in the list.  Not until I made a modification to another part of the UI that modified the internal view model did the button reenable.  After adding the code in the pull request, it works just fine.

The idea is that whenever the CommandManager believes an event occurred in the UI that it thinks the ability of a command to run has changed (such as the focus has changed, the content of a TextBox has changed, a new ListBoxItem was selected, etc), it will raise the RequerySuggested event, and all RoutedCommands are designed to respond to that event by raising the CanExecuteChanged event. Essentially, I extended Command to do the same.
